### ISSUE TITLE
cttmetrics is not reusable: cttmetrics.h is not installed in the target include folders

### ISSUE THREAD

**Issue by MathieuGirard** [Intel-Media-SDK/MediaSDK#2773]
cttmetrics is a neat library that could be reusable. Could we at least install de header file too?

https://github.com/Intel-Media-SDK/MediaSDK/blob/914f4789f4764f7af2c05a4443271cf0c1dd25f3/samples/metrics_monitor/CMakeLists.txt#L16-L19

The cttmetrics.so library gets installed but not the header file that defines the types and entry points.

Here is a quick path that would install in (/opt/intel/mediasdk/include/) probably not the best path (there is no samples header files path):
```diff
diff --git a/samples/metrics_monitor/CMakeLists.txt b/samples/metrics_monitor/CMakeLists.txt
index 9e643ee2..33d6b5e0 100755
--- a/samples/metrics_monitor/CMakeLists.txt
+++ b/samples/metrics_monitor/CMakeLists.txt
@@ -16,6 +16,7 @@ if( PKG_LIBDRM_FOUND AND
make_library( cttmetrics_static none static )

install( TARGETS cttmetrics LIBRARY DESTINATION ${MFX_SAMPLES_INSTALL_BIN_DIR} )
+ install( FILES include/cttmetrics.h DESTINATION ${CMAKE_INSTALL_FULL_INCLUDEDIR} )

# metrics_monitor sample
```

**Comment by dvrogozh** [comment]
No. Installing just headers is not enough. It will also require to install cttmetrics to LIBDIR and providing .pc file for it. Basically it would mean productization of the library which does not actually makes much sense. I believe that better way is to refer to the source code of cttmetrics library, borrow code examples and directly reuse in the application or write your own library. Basically cttmetrics is just educational purpose stuff to guide towards https://man7.org/linux/man-pages/man2/perf_event_open.2.html. See https://github.com/Intel-Media-SDK/MediaSDK/blob/master/samples/metrics_monitor/src/cttmetrics_i915_pmu.cpp
